projects/storefrontlib/src/cms-components/configurator/commons/config-attribute-footer/config-attribute-footer.component.ts

```diff
@@ -1,30 +1,41 @@
 import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
 import { Configurator, GenericConfigurator } from '@spartacus/core';
 import { ICON_TYPE } from '../../../misc/icon/icon.model';
+import { map } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+import { ConfigUtilsService } from '../service/config-utils.service';
 
 @Component({
   selector: 'cx-config-attribute-footer',
   templateUrl: './config-attribute-footer.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class ConfigAttributeFooterComponent {
-  constructor() {}
+  constructor(private configUtils: ConfigUtilsService) {}
+
   iconTypes = ICON_TYPE;
 
   @Input() attribute: Configurator.Attribute;
-  @Input() ownerType: GenericConfigurator.OwnerType;
+  @Input() owner: GenericConfigurator.Owner;
+  @Input() groupId: string;
 
-  showRequiredMessage(): boolean {
-    if (
-      this.ownerType === GenericConfigurator.OwnerType.CART_ENTRY &&
-      this.attribute.required &&
-      this.attribute.incomplete &&
-      this.attribute.uiType === Configurator.UiType.STRING &&
-      !this.attribute.userInput
-    ) {
-      return true;
-    }
-    return false;
+  showRequiredMessage(): Observable<boolean> {
+    return this.configUtils
+      .isCartEntryOrGroupVisited(this.owner, this.groupId)
+      .pipe(
+        map((result) => {
+          if (
+            result &&
+            this.attribute.required &&
+            this.attribute.incomplete &&
```

Comment:
This line is not needed, right? If an attribute is required and userInput is blank, we need to show the message. (Or am I wrong?)

Reply:
And if we merge this one _after_ the PR dealing with numeric attributes, we need to cover UiType.NUMERIC as well

Reply:
Good hint! I completely forgot about the numeric input validation.